cc-wrapper.sh

```diff
@@ -64,14 +64,20 @@ fi
 
 #echo "compiling to $dest..."
 
-escapedArgs='"-o" "${placeholder "out"}" "-c" '$(readlink -f "$dest.$ext")' '
+# compileFlags like "-x c++-header" must come before the source arguments,
+# otherwise the flags are ignored.
+# TODO ideally preserve the original order of arguments
```

Comment:
here is a start ...

https://github.com/milahu/gcc-options-parser/blob/master/gcc-options-parser.sh